cmd/uhc/cluster/list/list.go

```diff
@@ -220,11 +298,53 @@ func run(cmd *cobra.Command, argv []string) {
 
 		// Display the fetched page:
 		response.Items().Each(func(cluster *v1.Cluster) bool {
-			// Get cluster API path
+
+			// If the custom columns flag was raised,
+			// populate cluster array with custom columns:
+			if args.columns != "" {
+
+				// String to output marshal -
+				// Map used to parse Cluster data -
+				// Writer to body variable:
+				var body string
+				var jsonBody map[string]interface{}
+				boddyBuffer := bytes.NewBufferString(body)
+
+				// Write Cluster data to body variable:
+				err := v1.MarshalCluster(cluster, boddyBuffer)
+				if err != nil {
+					fmt.Fprintf(os.Stderr, "Failed to Marshal Cluster into byte buffer: %s\n", err)
+					os.Exit(1)
+				}
+
+				// Get JSON from Cluster bytes
+				err = json.Unmarshal(boddyBuffer.Bytes(), &jsonBody)
+				if err != nil {
+					fmt.Fprintf(os.Stderr, "Failed to turn Cluster bytes into JSON map: %s\n", err)
+					os.Exit(1)
+				}
+
+				// Loop through wanted columns and populate a cluster instance
+				iter := strings.Split(args.columns, ",")
+				thisCluster := []string{}
+				for _, element := range iter {
+					elementPointer := &element
+					_ = findMapValue(jsonBody, elementPointer)
+					thisCluster = append(thisCluster, *elementPointer)
+					thisCluster = updateRowPad([]int{args.padding}, thisCluster)
+				}
+
+				// Print current cluster:
+				prettyPrint(thisCluster)
+				return true
+			}
+			// Default cluster columns:
+
+			// Get cluster API path:
 			api := cluster.API()
 			apiURL := api.URL()
 
-			// Populate this cluster instance
+			// Populate this cluster instance:
```

Comment:
As I said I think we should not preserve this special case. We should instead have a default value for the `columns` option.